### DISCUSSION TITLE
License confusion: "No attribution required" vs. MIT

### DISCUSSION THREAD

**Discussion by denisw** [shadcn-ui/ui#2339] [Q&A]
I just learned about `shadcn/ui` and find the concept very interesting. However, I think there is a mismatch between the FAQ and the `LICENSE.md` file regarding the license terms.

The FAQ in the docs says (emphasis mine):

> Yes. Free to use for personal and commercial projects. __No attribution required.__

However, the license file points at the MIT License, whose one and only condition is exactly attribution:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

If "no attribution required" is the intention, then perhaps another license needs to be picked to remove the confusion. For example, the [Zero-Clause BSD (BSD-0) License](https://opensource.org/license/0bsd/) could be a fit.

**Comment by shadcn** [comment]
Interestin (and nice catch). I'll look into the licensing and get back on this.

**Reply by kotx** [reply]
Would like an update on this too. Thanks!

**Reply by lukaw3d** [reply]
"No attribution required" line was removed in https://github.com/shadcn-ui/ui/pull/6605/files#diff-cbbe6d5afa3e2e0bdafa6c037fcf61a07770b86c152dc63ab52e3a105edbbf9dL59

**Comment by ClosetGeek-Git** [comment]
Not a lawyer, but heres the gist. Attribution generally means that you must state the use of the software in a way that users can see. This can be a ‘About’ drop down, a notice in the footer, a notice that displays their license during installation, etc. MIT on the other hand just requires you to retain existing copyright notices where they exist within the original code. You also must distribute the original LICENSE file if you add the software as part of a larger work to be distributed as a software product. The passing of this license file is not attribution, but forwarding of license as stated in the terms. Remember, license is not simply a type of file or notice. _It is a legal transfer of permission regarding how software can be used_. 

If one buys a used DVD copy of Windows Server it doesn’t mean they have the right to use it, they just own the plastic disk. Where this gets sticky is if the seller implies they are giving you the right to use the software as part of them passing the software to you. The buyer may find that they can't use the product without obtaining a license - not a file per se, but legal permission. Additionally, for the distributor it can quickly become a _crime_, especially if the distributor clearly implies they can use the software without worrying about additional license.

So no, you do not have to attribute. You just have to make the license available and cannot remove the license from where it may exist in code files.

**Comment by ClosetGeek-Git** [comment]
I just found a second potential issue. I've read that shadcn-ui is "built on top of" Radix. I'm not sure of the extent of this, but implying users can redistribute freely could be an issue if shadcn-ui/ui is bound to any form of license from Radix.

**Comment by sourman** [comment]
would love to hear more on this

**Comment by cbix** [comment]
Some libraries will leave a license header as comment in your bundle. I'd expect if attribution was required shadcn/ui would put similar comments in the installed components code.

**Comment by svwezel** [comment]
Or this licence? https://github.com/aws/mit-0
